### PR TITLE
[s2s] add config params like Dropout in Seq2SeqTrainingArguments

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

1. Adds `config` params (`encoder_layerdrop`, `decoder_layerdrop`, `dropout`, `attention_dropout`) in `Seq2SeqTrainingArguments`
2. Fix T5 warnings (don't pass src_lang, tgt_lang args to `T5Tokenizer`)
3. Correct `vocab_size` for `FSMT`.
4. Fix `test_finetune_trainer_slow`
5. minor code cleanup in `Seq2SeqTrainer`

@sshleifer

### CODE REVIEW

File: examples/seq2seq/test_finetune_trainer.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def test_finetune_trainer():
 def test_finetune_trainer_slow():
     # TODO(SS): This will fail on devices with more than 1 GPU.
     # There is a missing call to __init__process_group somewhere
-    output_dir = run_trainer(eval_steps=2, max_len="32", model_name=MARIAN_MODEL, num_train_epochs=3)
+    output_dir = run_trainer(eval_steps=2, max_len="128", model_name=MARIAN_MODEL, num_train_epochs=3)
```
Comment (patil-suraj) [inline examples/seq2seq/test_finetune_trainer.py:29]: slow test fails when `max_len` is 32

**Review by sshleifer — COMMENTED** [review]
LGTM! Want me to merge?

**Comment by patil-suraj** [comment]
Yes, will add lr scheduler in separate PR

File: examples/seq2seq/test_finetune_trainer.py (hunk 1 above)
Comment (sshleifer) [reply examples/seq2seq/test_finetune_trainer.py:29]: my bad, passed on my machine somehow.

**Review by sshleifer — APPROVED** [review]

**Comment by patil-suraj** [comment]
@sshleifer  anything missing in this PR ?

**Pull request merged**